src/ocamlorg_package/lib/ocamlorg_package.ml

```diff
@@ -375,28 +380,32 @@ let search_index ~kind t =
 
 module Documentation_status = Documentation_status
 
-let documentation_status ~kind t : Documentation_status.t option Lwt.t =
+let documentation_status ~kind state t : Documentation_status.t option Lwt.t =
   let open Lwt.Syntax in
   let package_url =
     package_url ~kind (Name.to_string t.name) (Version.to_string t.version)
   in
   let url = package_url ^ "status.json" in
-  let* content = http_get url in
-  let status =
-    match content with
-    | Ok s ->
-        Some (s |> Yojson.Safe.from_string |> Documentation_status.of_yojson)
-    | _ -> None
-  in
-  Lwt.return status
+  match DocumentationStatusMap.find_opt url state.doc_stats with
+  | Some status -> Lwt.return status
+  | None ->
+      let+ content = http_get url in
+      let status =
+        match content with
+        | Ok s ->
+            Some (s |> Yojson.Safe.from_string |> Documentation_status.of_yojson)
+        | _ -> None
+      in
+      state.doc_stats <- DocumentationStatusMap.add url status state.doc_stats;
```

Comment:
I see, this caches by URL. I suppose that caching by `Ocamlorg_package.Name.t` and `Version.t` would probably be more obvious.

Reply:
like in the `packages` field, right?

Reply:
I did it via the URL because we need to take the `kind` also into account, and the URL already encodes that info. Do you have and idea/example of that?